webhook_to_fedora_messaging_messages/forgejo/v1.py

```diff
@@ -0,0 +1,78 @@
+# SPDX-FileCopyrightText: 2024 Contributors to the Fedora Project
+#
+# SPDX-License-Identifier: LGPL-3.0-or-later
+
+import typing
+import uuid
+
+from ..base import Webhook2FedMsgBase
+from .utils import summarize_repository_event
+
+
+class ForgejoMessageV1(Webhook2FedMsgBase):
+    @property
+    def app_name(self):
+        return "Forgejo"
+
+    @property
+    def target_id(self):
+        return self.body["repository"]["id"]
+
+    @property
+    def signature(self):
+        return hashlib.sha1(str(self.body).encode(encoding="utf-8")).hexdigest()
```

Comment:
The body is a dictionary, and the order of the keys is not guaranteed to be consistent when serializing it. As a result, you may get a different signature for different "calls" of this property. You can use `json.dumps(value, sort_keys=True)` to get a sorted text representation of the dict.